force-app/main/default/classes/OpportunityTriggerHandler.cls

```diff
@@ -0,0 +1,180 @@
+ *  - afterDelete: Alerts owner of the Oppotunity when it is deleted
+ *  - afterUndelete: Reassigns undeleted Opportunities to a new Primary Contact
+ * 
+ * @author Ilana Shotkin
+*/
+
+public with sharing class OpportunityTriggerHandler extends TriggerHandler {
+
+    // Class variables
+    private List<Opportunity> newOppList;
+    private List<Opportunity> oldOppList;
+    private Map<Id, Opportunity> newOppMap;
+    private Map<Id, Opportunity> oldOppMap;
+
+    // Class Constructor
+    public OpportunityTriggerHandler(List<Opportunity> newOppList, List<Opportunity> oldOppList, Map<Id, Opportunity> newOppMap, Map<Id, Opportunity> oldOppMap) {
+        this.newOppList = newOppList;
+        this.oldOppList = oldOppList;
+        this.newOppMap = newOppMap;
+        this.oldOppMap = oldOppMap;
+    }
+    
+    /* Before Save methods */
+
+    
+    /* 
+     * beforeInsert handler method:
+     * 
+     * Set a default Type of "New Customer" if no Type is provided upon creation
+     */
+    public override void beforeInsert() {
+        // Set default Type for new Opportunities
+        for (Opportunity opp : newOppList) {
+            if (opp.Type == null){
+                opp.Type = 'New Customer';
+            }
+        }  
+    }
+
+    /* 
+     * beforeUpdate handler method:
+     * 
+     * Validate that the Amount is not under 5000. Upend Stage changes to the Desription. If the Primary Contact is not provided, 
+     * set the Primcary COntact to the Opportunity's Account's CEO
+     */
+    public override void beforeUpdate() {
+        // Create an empty map to store Contacts with their AccountId
+        Map<Id, Contact> accountIdToContact = new Map<Id, Contact>();
+        
+        // When an opportunity is updated, validate that the amount is greater than 5000.
+        for (Opportunity opp : newOppList){
+            if(opp.Amount < 5000){
+                opp.addError('Opportunity amount must be greater than 5000');
+
+            } else {
+                 // Only update the Opportunities than have a non-null StageName and whose StageName has changed
+                if (opp.StageName != null && opp.StageName != oldOppMap.get(opp.Id).StageName){ 
+                opp.Description = opp.Description + '\n Stage Change:' + opp.StageName + ':' + DateTime.now().format();
+                }   
+            }
+        }
+
+        // Get the Primary Contact for all Opportunities that are missing them
+        Map<Id, Id> oppsWithPrimaryContact = OpportunityUtils.setPrimaryContact(newOppList, 'CEO');
+
+        // Set the Primary Contact on triggered Opportunities that need one
+        if (!oppsWithPrimaryContact.isEmpty()) {
+            for (Opportunity opp : newOppList) {
+                if (opp.Primary_Contact__c == null && oppsWithPrimaryContact.containsKey(opp.Id)){
+                    opp.Primary_Contact__c = oppsWithPrimaryContact.get(opp.Id);
+                }
```

Comment:
Why not just do this in your setPrimaryContact method? Either update the opportunities in place or return them back instead of the Map<Id, Id>